pkg/servicetest/servicetest_test.go

```diff
@@ -136,3 +153,46 @@ func TestServices(t *testing.T) {
 		}
 	}
 }
+
+// checkHTTP will attempt to communicate with a service over HTTP
+func checkHTTP(t *testing.T, container docker.APIContainers, targetPort int64, path string) {
+	t.Log("Checking HTTP communication")
+	assert := asrt.New(t)
+
+	for _, port := range container.Ports {
+		if port.PrivatePort == targetPort && port.PublicPort != 0 {
+			address := fmt.Sprintf("http://127.0.0.1:%d%s", port.PublicPort, path)
+			t.Logf("Checking %s for 200 status", address)
+
+			o := util.NewHTTPOptions(address)
+			o.ExpectedStatus = 200
+			o.Timeout = 30
+			runcheckErr := util.EnsureHTTPStatus(o)
+			assert.NoError(runcheckErr)
+
+			return
+		}
+	}
+
+	t.Fatalf("Unable to find target port: %d", targetPort)
+}
+
+// checkTCP will attempt to communicate with a service over TCP
+func checkTCP(t *testing.T, container docker.APIContainers, targetPort int64, path string) {
+	t.Log("Checking TCP communication")
+	assert := asrt.New(t)
+
+	for _, port := range container.Ports {
+		if port.PrivatePort == targetPort {
+			address := fmt.Sprintf("127.0.0.1:%d%s", port.PublicPort, path)
```

Comment:
As with the other one, this tests from the host's perspective, but may not be an adequately valid test because it's not testing from the web container's perspective.